pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -537,7 +537,16 @@ func (e *Controller) syncService(ctx context.Context, key string) error {
 		_, err = e.client.CoreV1().Endpoints(service.Namespace).Create(ctx, newEndpoints, metav1.CreateOptions{})
 	} else {
 		// Pre-existing
-		_, err = e.client.CoreV1().Endpoints(service.Namespace).Update(ctx, newEndpoints, metav1.UpdateOptions{})
+		var updatedEndpoints *v1.Endpoints
+		updatedEndpoints, err = e.client.CoreV1().Endpoints(service.Namespace).Update(ctx, newEndpoints, metav1.UpdateOptions{})
+		// a webhook could potentially cause endpoints update operation became noop and return same resourceVersion.
```

Comment:
```suggestion
		// a webhook could potentially cause endpoints update operation to become noop and return same resourceVersion.
```

Reply:
thanks, i'm see how to test this change without create a real webhook. an UT shall be enough right?

Reply:
let's generalize the comment
```suggestion
		// there are some operations (webhooks, truncated endpoints, ...) that can potentially cause endpoints updates became noop and return the same resourceVersion.
		// Ref: https://issues.k8s.io/127370 , https://issues.k8s.io/126578
```